service/service.go

```diff
@@ -473,24 +482,38 @@ func (s *Service) Submit(
 
 	type response struct {
 		round string
+		done  <-chan error
 		err   error
 		end   time.Time
 	}
-	done := make(chan response, 1)
+	cmdFinished := make(chan response, 1)
 	s.commands <- func(s *Service) {
```

Comment:
maybe add a histogram with buckets from 0.1s to 10s to track latency distribution for submissions. it will allow to track queue saturation, and based on that max size and wait time can be tuned

Reply:
Bumped grpc metrics to go-grpc-middleware and configured histogram for GRPC latency to 20 exponential buckets from 0.0001 with 2x step which results in:
```
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0001"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0002"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0004"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0008"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0016"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0032"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0064"} 0
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0128"} 387
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0256"} 5194
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.0512"} 17538
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.1024"} 32896
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.2048"} 75270
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.4096"} 95454
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="0.8192"} 99913
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="1.6384"} 100000
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="3.2768"} 100000
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="6.5536"} 100000
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="13.1072"} 100000
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="26.2144"} 100000
grpc_server_handling_seconds_bucket{grpc_method="Submit",grpc_service="rpc.api.v1.PoetService",grpc_type="unary",le="52.4288"} 100000
```